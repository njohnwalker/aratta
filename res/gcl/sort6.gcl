// in-place sorts 6 integer variables
@req()
// phantom variables for invariant
p0, p1, p2, p3, p4, p5
    := x0, x1, x2, x3, x4, x5
do @inv ()
[]  x0 > x1 | x1 > x2 | x2 > x3 | x3 > x4 | x4 > x5 ->
        if x0 > x1 -> x0, x1 := x1, x0 ;
        [] x2 > x3 -> x2, x3 := x3, x2 ;
        [] x4 > x5 -> x4. x5 := x5, x4 ;
        fi
        if x1 > x2 -> x1, x2 := x2, x1 ;
        [] x3 > x4 -> x3, x4 := x4, x3 ;
        fi
od
@ens( x0 <= x1
    & x1 <= x2
    & x2 <= x3
    & x3 <= x4
    & x4 <= x5
)