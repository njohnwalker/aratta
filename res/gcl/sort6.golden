res/gcl/sort6.gcl:6:1:
  |
6 | do @inv ()
  | ^
unexpected 'd'
expecting '%', '*', '+', ',', '-', '/', or ';'
